proc/yield/yield.go

```diff
@@ -39,13 +37,9 @@ func (p *Proc) Pull() (zbuf.Batch, error) {
 				recs = append(recs, *out.Copy())
 			}
 		}
-		batch.Unref()
+		defer batch.Unref()
 		if len(recs) > 0 {
-			return zbuf.NewArray(recs), nil
+			return zbuf.NewBatch(batch, recs), nil
 		}
```

Comment:
Best to avoid deferring inside a loop like this since it will require allocations if the trip count is high.
```go
		if len(recs) > 0 {
			defer batch.Unref()
			return zbuf.NewBatch(batch, recs), nil
		}
		defer batch.Unref()
```

Reply:
I looked a bit at the implementation and [`defer` in a loop requires an allocation regardless of trip count](https://github.com/golang/go/blob/2580d0e08d5e9f979b943758d3c49877fb2324cb/src/runtime/panic.go#L14-L19).

Reply:
It is the same. 🤦

Here's what I meant to suggest:
```go
		if len(recs) > 0 {
			defer batch.Unref()
			return zbuf.NewBatch(batch, recs), nil
		}
		batch.Unref()
```